tests/test_downloader.py

```diff
@@ -101,6 +103,25 @@ def test_download_server_error(
         # assert exception catched
         assert isinstance(e.value.__cause__, inject_requests_err)
 
+    def test_raise_streaming_error_on_incompleted_download(self, tmp_path: Path):
+        from otaclient.app.downloader import Downloader
+
+        _downloader = Downloader()
+        _target_path = tmp_path / self.TEST_FILE
+
+        with pytest.raises(ChunkStreamingError) as e:
+            _downloader.download(
+                self.TEST_FILE,
+                _target_path,
+                digest=self.TEST_FILE_SHA256,
+                size=random.randint(
```

Comment:
I'd like to remove randomness from the test.

Reply:
`size=self.TEST_FILE_SIZE // 2` is enough.

Reply:
OK!

Reply:
Updated in [13ce31b](https://github.com/tier4/ota-client/pull/157/commits/13ce31bf903a3731d7b66fd32783579028517ae9).